sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/PartitionContext.cs

```diff
@@ -29,21 +32,76 @@ public class PartitionContext
         ///
         public string PartitionId { get; }
 
+        /// <summary>
+        ///   The identifier of the associated <see cref="EventProcessor{T}" /> instance.
+        /// </summary>
+        ///
+        public string OwnerIdentifier { get; }
+
+        /// <summary>
+        ///   Interacts with the storage system, dealing with the creation of checkpoints.
+        /// </summary>
+        ///
+        private PartitionManager Manager { get; }
+
         /// <summary>
         ///   Initializes a new instance of the <see cref="PartitionContext"/> class.
         /// </summary>
         ///
         /// <param name="eventHubName">The name of the specific Event Hub this context is associated with, relative to the Event Hubs namespace that contains it.</param>
         /// <param name="consumerGroup">The name of the consumer group this context is associated with.</param>
         /// <param name="partitionId">The identifier of the Event Hub partition this context is associated with.</param>
+        /// <param name="partitionManager">Interacts with the storage system, dealing with the creation of checkpoints.</param>
+        /// <param name="ownerIdentifier">The identifier of the associated <see cref="EventProcessor{T}" /> instance.</param>
         ///
         internal PartitionContext(string eventHubName,
                                   string consumerGroup,
-                                  string partitionId)
+                                  string partitionId,
+                                  string ownerIdentifier,
+                                  PartitionManager partitionManager)
         {
             EventHubName = eventHubName;
             ConsumerGroup = consumerGroup;
             PartitionId = partitionId;
+            OwnerIdentifier = ownerIdentifier;
+            Manager = partitionManager;
+        }
+
+        /// <summary>
+        ///   Updates the checkpoint using the given information for the associated partition and consumer group in the chosen storage service.
+        /// </summary>
+        ///
+        /// <param name="eventData">The event containing the information to be stored in the checkpoint.</param>
+        ///
+        /// <returns>A task to be resolved on when the operation has completed.</returns>
+        ///
+        public Task UpdateCheckpointAsync(EventData eventData) => UpdateCheckpointAsync(eventData.Offset.Value, eventData.SequenceNumber.Value);
```

Comment:
What if offset or sequence number are `null`?

Reply:
I'll add some guards and let them throw in case it is `null`. This kind of situation can only happen when the user creates an `EventData` instance and tries to use it as `Checkpoint` instead of using an `EventData` received from the Event Hubs service.

This method can only be called from one of the `PartitionProcessor`'s methods (`Initialize`, `Close`, `ProcessEvents`, `ProcessError`) or constructor. Let the exception bubble up through them. We should already be handling these kinds of exceptions (or delegating them to our nonexistent error handler).